internal/middleware/util.go

```diff
@@ -25,3 +31,135 @@ func GetFromContext[V any, I any](i I) func(context.Context) (*V, error) {
 
 	})
 }
+
+func NormalizeResourceType(resourceType string) string {
```

Comment:
I might be missing something obvious, but, wouldn't it be "easier" if we just keep the filepath -as-is- as the key for the cache? I've seen a bunch of helpers around this key convention, and I'm not sure I understand if it's needed?

Reply:
The challenge with the `notifications_integration` is that the resource type submitted by the notification team is "resource_type": "notifications/integration," which is different from other types. During the preload logic, a schema cache with key notifications/integration converted to notifications_integration is created. That's why helper function is used. Otherwise we need to add some if/else block for notifications